static/app/views/organizationGroupDetails/groupEventDetails/index.tsx

```diff
@@ -31,53 +29,48 @@ type Props = RouteComponentProps<
   organization: Organization;
   project: Project;
   selection: PageFilters;
-};
+}
 
-type State = typeof OrganizationEnvironmentsStore['state'];
+export function GroupEventDetailsContainer(props: GroupEventDetailsProps) {
+  const [state, setState] = useState<typeof OrganizationEnvironmentsStore['state']>(
+    OrganizationEnvironmentsStore.get()
+  );
 
-export class GroupEventDetailsContainer extends Component<Props, State> {
-  state = OrganizationEnvironmentsStore.get();
+  useEffect(() => {
+    const unsubscribe = OrganizationEnvironmentsStore.listen(data => {
+      setState(data);
+    }, undefined);
```

Comment:
Can you `useLegacyStore` here?

Reply:
Then you should be able to totally get rid of the `useState` I think, since this component is really just driven by the store right?

Reply:
Nice, yeah. I didnt know about that hook, thanks for pointing it out!